ggml.c

```diff
@@ -7568,8 +7573,15 @@ static void ggml_compute_forward_mul_mat_f32(
 
                 // copy data to host
                 CUDA_CHECK(cudaMemcpyAsync(d, d_D, sizeof(float) * d_ne, cudaMemcpyDeviceToHost, g_cudaStream));
-#else
+#elif defined(GGML_USE_CLBLAST)
                 // zT = y * xT
+                ggml_cl_sgemm_wrapper(CLBlastLayoutRowMajor, CLBlastTransposeNo, CLBlastTransposeYes,
+                        ne11, ne01, ne10,
+                        1.0f,    y, ne10,
+                                 x, ne10,
+                        0.0f,    d, ne01,
+                        params->type);
```

Comment:
Passing `params->type` here seems wrong, as this is of type `ggml_task_type`. You may want to pass something like `src0->type` or `GGML_TYPE_F32`, and then in `ggml_cl_sgemm_wrapper` check that the type is one of the supported values, to prevent it from failing silently when a new quantized type is added.

Reply:
Good catch.